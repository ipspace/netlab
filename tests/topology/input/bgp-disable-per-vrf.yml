provider: clab

defaults.device: srlinux

vrfs:
  red:

  blue:
    bgp: False # Disable BGP sessions in 'blue' VRF

vlans:
  red:
    mode: route
    vrf: red
  blue:
    mode: route
    vrf: blue

module: [ vlan, vrf, bgp ]

nodes:
  r1:
    bgp.as: 65100
  r2:
    bgp.as: 65000
    vrfs:
      red:
        bgp.router_id: 172.31.0.1
      blue:
        bgp.router_id: 172.32.0.2
  r3:
    bgp.as: 65101

links:
- r1:
  r2:
  vlan.trunk: [ red, blue ]
- r2:
  r3:
  vlan.trunk: [ red, blue ]
