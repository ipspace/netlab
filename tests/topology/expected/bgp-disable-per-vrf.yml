bgp:
  advertise_loopback: true
  community:
    ebgp:
    - standard
    ibgp:
    - standard
    - extended
  next_hop_self: true
groups:
  as65000:
    members:
    - r2
  as65100:
    members:
    - r1
  as65101:
    members:
    - r3
input:
- topology/input/bgp-disable-per-vrf.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: ethernet-1/1
    node: r1
    vlan:
      trunk:
        blue: {}
        red: {}
  - ifindex: 1
    ifname: ethernet-1/1
    node: r2
    vlan:
      trunk:
        blue: {}
        red: {}
  linkindex: 1
  node_count: 2
  prefix: {}
  role: external
  type: p2p
  vlan:
    trunk:
      blue: {}
      red: {}
- interfaces:
  - ifindex: 2
    ifname: ethernet-1/2
    node: r2
    vlan:
      trunk:
        blue: {}
        red: {}
  - ifindex: 1
    ifname: ethernet-1/1
    node: r3
    vlan:
      trunk:
        blue: {}
        red: {}
  linkindex: 2
  node_count: 2
  prefix: {}
  role: external
  type: p2p
  vlan:
    trunk:
      blue: {}
      red: {}
module:
- vlan
- bgp
- vrf
name: input
nodes:
  r1:
    af:
      ipv4: true
      vpnv4: true
    bgp:
      advertise_loopback: true
      as: 65100
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      neighbors: []
      next_hop_self: true
      router_id: 10.0.0.1
    box: ghcr.io/nokia/srlinux
    clab:
      kind: srl
      type: ixrd2
    device: srlinux
    hostname: clab-input-r1
    id: 1
    interfaces:
    - clab:
        name: e1-1
      ifindex: 1
      ifname: ethernet-1/1
      linkindex: 1
      name: r1 -> r2
      neighbors:
      - ifname: ethernet-1/1
        node: r2
      role: external
      subif_index: 2
      type: p2p
    - bridge_group: 1
      ifindex: 2
      ifname: ethernet-1/1.1001
      ipv4: 172.16.2.1/24
      name: r1 -> r2
      neighbors:
      - ifname: ethernet-1/1.1001
        ipv4: 172.16.2.2/24
        node: r2
        vrf: blue
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
      vrf: blue
    - bridge_group: 2
      ifindex: 3
      ifname: ethernet-1/1.1000
      ipv4: 172.16.3.1/24
      name: r1 -> r2
      neighbors:
      - ifname: ethernet-1/1.1000
        ipv4: 172.16.3.2/24
        node: r2
        vrf: red
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
      vrf: red
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: mgmt0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - vlan
    - bgp
    - vrf
    name: r1
    vlan:
      max_bridge_group: 2
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
        vrf: blue
      red:
        bridge_group: 2
        id: 1000
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      red:
        af:
          ipv4: true
        bgp:
          neighbors:
          - as: 65000
            ifindex: 3
            ipv4: 172.16.3.2
            name: r2
            type: ebgp
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  r2:
    af:
      ipv4: true
      vpnv4: true
    bgp:
      advertise_loopback: true
      as: 65000
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      neighbors: []
      next_hop_self: true
      router_id: 10.0.0.2
    box: ghcr.io/nokia/srlinux
    clab:
      kind: srl
      type: ixrd2
    device: srlinux
    hostname: clab-input-r2
    id: 2
    interfaces:
    - clab:
        name: e1-1
      ifindex: 1
      ifname: ethernet-1/1
      linkindex: 1
      name: r2 -> r1
      neighbors:
      - ifname: ethernet-1/1
        node: r1
      role: external
      subif_index: 2
      type: p2p
    - clab:
        name: e1-2
      ifindex: 2
      ifname: ethernet-1/2
      linkindex: 2
      name: r2 -> r3
      neighbors:
      - ifname: ethernet-1/1
        node: r3
      role: external
      subif_index: 2
      type: p2p
    - bridge_group: 1
      ifindex: 3
      ifname: ethernet-1/1.1001
      ipv4: 172.16.2.2/24
      name: r2 -> r1
      neighbors:
      - ifname: ethernet-1/1.1001
        ipv4: 172.16.2.1/24
        node: r1
        vrf: blue
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
      vrf: blue
    - bridge_group: 2
      ifindex: 4
      ifname: ethernet-1/1.1000
      ipv4: 172.16.3.2/24
      name: r2 -> r1
      neighbors:
      - ifname: ethernet-1/1.1000
        ipv4: 172.16.3.1/24
        node: r1
        vrf: red
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
      vrf: red
    - bridge_group: 1
      ifindex: 5
      ifname: ethernet-1/2.1001
      ipv4: 172.16.4.2/24
      name: r2 -> r3
      neighbors:
      - ifname: ethernet-1/1.1001
        ipv4: 172.16.4.3/24
        node: r3
        vrf: blue
      parent_ifindex: 2
      parent_ifname: ethernet-1/2
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
      vrf: blue
    - bridge_group: 2
      ifindex: 6
      ifname: ethernet-1/2.1000
      ipv4: 172.16.5.2/24
      name: r2 -> r3
      neighbors:
      - ifname: ethernet-1/1.1000
        ipv4: 172.16.5.3/24
        node: r3
        vrf: red
      parent_ifindex: 2
      parent_ifname: ethernet-1/2
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
      vrf: red
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: mgmt0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    module:
    - vlan
    - bgp
    - vrf
    name: r2
    vlan:
      max_bridge_group: 2
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
        vrf: blue
      red:
        bridge_group: 2
        id: 1000
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        bgp:
          neighbors:
          - as: 65100
            ifindex: 3
            ipv4: 172.16.2.1
            name: r1
            type: ebgp
          - as: 65101
            ifindex: 5
            ipv4: 172.16.4.3
            name: r3
            type: ebgp
          router_id: 172.32.0.2
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      red:
        af:
          ipv4: true
        bgp:
          neighbors:
          - as: 65100
            ifindex: 4
            ipv4: 172.16.3.1
            name: r1
            type: ebgp
          - as: 65101
            ifindex: 6
            ipv4: 172.16.5.3
            name: r3
            type: ebgp
          router_id: 172.31.0.1
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  r3:
    af:
      ipv4: true
      vpnv4: true
    bgp:
      advertise_loopback: true
      as: 65101
      community:
        ebgp:
        - standard
        ibgp:
        - standard
        - extended
      neighbors: []
      next_hop_self: true
      router_id: 10.0.0.3
    box: ghcr.io/nokia/srlinux
    clab:
      kind: srl
      type: ixrd2
    device: srlinux
    hostname: clab-input-r3
    id: 3
    interfaces:
    - clab:
        name: e1-1
      ifindex: 1
      ifname: ethernet-1/1
      linkindex: 2
      name: r3 -> r2
      neighbors:
      - ifname: ethernet-1/2
        node: r2
      role: external
      subif_index: 2
      type: p2p
    - bridge_group: 1
      ifindex: 2
      ifname: ethernet-1/1.1001
      ipv4: 172.16.4.3/24
      name: r3 -> r2
      neighbors:
      - ifname: ethernet-1/2.1001
        ipv4: 172.16.4.2/24
        node: r2
        vrf: blue
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1001
        mode: route
        routed_link: true
      vrf: blue
    - bridge_group: 2
      ifindex: 3
      ifname: ethernet-1/1.1000
      ipv4: 172.16.5.3/24
      name: r3 -> r2
      neighbors:
      - ifname: ethernet-1/2.1000
        ipv4: 172.16.5.2/24
        node: r2
        vrf: red
      parent_ifindex: 1
      parent_ifname: ethernet-1/1
      role: external
      type: vlan_member
      virtual_interface: true
      vlan:
        access_id: 1000
        mode: route
        routed_link: true
      vrf: red
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: mgmt0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    module:
    - vlan
    - bgp
    - vrf
    name: r3
    vlan:
      max_bridge_group: 2
    vlans:
      blue:
        bridge_group: 1
        id: 1001
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
        vrf: blue
      red:
        bridge_group: 2
        id: 1000
        mode: route
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vrf: red
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        rd: '65000:2'
        vrfidx: 101
      red:
        af:
          ipv4: true
        bgp:
          neighbors:
          - as: 65000
            ifindex: 3
            ipv4: 172.16.5.2
            name: r2
            type: ebgp
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
provider: clab
vlans:
  blue:
    id: 1001
    mode: route
    prefix:
      allocation: id_based
      ipv4: 172.16.1.0/24
    vrf: blue
  red:
    id: 1000
    mode: route
    prefix:
      allocation: id_based
      ipv4: 172.16.0.0/24
    vrf: red
vrf:
  as: 65000
vrfs:
  blue:
    bgp: false
    export:
    - '65000:2'
    id: 2
    import:
    - '65000:2'
    rd: '65000:2'
  red:
    export:
    - '65000:1'
    id: 1
    import:
    - '65000:1'
    rd: '65000:1'
